include/zephyr/drivers/charger.h

```diff
@@ -84,6 +98,66 @@ enum charger_status {
 	CHARGER_STATUS_FULL,
 };
 
+/**
+ * @brief Charge algorithm types
+ */
+enum charger_charge_type {
+	/** Charge type is unknown */
+	CHARGER_CHARGE_TYPE_UNKNOWN = 0,
```

Comment:
As we discussed offline, let's try to make these and the other new enum values more explicitly defined. Probably starting with fewer of them too and we can add as needed.